### PR TITLE
Enhance coding checks

### PR DESCRIPTION
**Pull request by guidomodarelli**
### Description
We want to improve our code standards, therefore we will implement additional checks in our development process.

- Update plugins within `devDependencies`
- Change the rule `eslint-plugin-filenames-simple` to `unicorn/filename-case`. The first rule would mark the entire file in red, and now it only marks the first line of the file and also suggests what the new name of the file should be.
- Modify the `tsconfig.json` file to incorporate the `public`, `common`, and `server` directories from the plugins.
- Move the `devDependencies` to the top of the project.
- Move `eslintrc` to the top of the project.
- Update global `defaultFormatter`.
- Update global `defaultFormatter` to `shellscript`.
- Improve `eslint` and `prettier` workflows.
- Remove unnecessary steps from `pull_request_template`
- Add `pre-commit` hook to run `eslint` and `prettier` checks
- Eslint rules:
  - `react/sort-comp`: When creating React components it is more convenient to always follow the same organisation for method order to help you easily find lifecycle methods, event handlers, etc.
  - `@typescript-eslint/member-ordering`: This rule aims to standardize the way classes, interfaces, and type literals are structured and ordered. A consistent ordering of fields, methods and constructors can make code easier to read, navigate, and edit.
  - `@stylistic/*`
    - `@stylistic/padding-line-between-statements`: This rule requires or disallows blank lines between the given 2 kinds of statements. Properly blank lines help developers to understand the code.
    - `@stylistic/no-multiple-empty-lines`: Some developers prefer to have multiple blank lines removed, while others feel that it helps improve readability. Whitespace is useful for separating logical sections of code, but excess whitespace takes up more of the screen.
    - `@stylistic/lines-between-class-members`: This rule improves readability by enforcing lines between class members. It will not check empty lines before the first member and after the last member, since that is already taken care of by padded-blocks.
  - `unicorn/*`
    - `unicorn/custom-error-definition`: Enforces the only valid way of Error subclassing. It works with any super class that ends in Error.
    - `unicorn/better-regex`: Improve regexes by making them shorter, consistent, and safer.
    - `unicorn/prevent-abbreviations`: Using complete words results in more readable code. Not everyone knows all your abbreviations. Code is written only once, but read many times.
  - `@typescript-eslint/*`
    - `@typescript-eslint/method-signature-style`: Enforce using a particular method signature syntax.TypeScript provides two ways to define an object/interface function property.
    - `@typescript-eslint/naming-convention`: Enforce naming conventions for everything across a codebase. Enforcing naming conventions helps keep the codebase consistent, and reduces overhead when thinking about how to name a variable. Additionally, a well-designed style guide can help communicate intent, such as by enforcing all private properties begin with an `_`, and all global-level constants are written in `UPPER_CASE`.

 
### Issues Resolved
https://github.com/wazuh/wazuh-dashboard-plugins/issues/7170

### Check List
- [ ] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by chantal-kelm — DISMISSED** [review]
CR: ✅ 
TEST: ✅ 

Captura de pantalla 2024-11-27 a la(s) 1 32 14 p  m
Captura de pantalla 2024-11-27 a la(s) 1 32 27 p  m
Captura de pantalla 2024-11-27 a la(s) 1 39 03 p  m

**Review by Desvelao — DISMISSED** [review]
LGTM!

File: docs/development/README.md
Comment (yenienserrano) [inline docs/development/README.md:1]: You can specify that it does not have to be inside the plugins folders because it would take another package.json

**Review by yenienserrano — APPROVED** [review]

**Review by yenienserrano — APPROVED** [review]
LGTM

**Review by Desvelao — APPROVED** [review]
LGTM!

**Pull request merged**